lua/tsc/init.lua

```diff
@@ -105,17 +104,44 @@ M.run = function()
     vim.defer_fn(notify, 125)
   end
 
+  local function notify_watch_mode()
+    notify_record =
+      vim.notify("👀 Watching your project for changes, kick back and relax 🚀", nil, get_notify_options())
+  end
+
   if config.enable_progress_notifications then
-    notify()
+    if config.flags.watch then
+      notify_watch_mode()
+    else
+      notify()
+    end
   end
 
   local function on_stdout(_, output)
     local result = utils.parse_tsc_output(output)
-
     errors = result.errors
     files_with_errors = result.files
 
-    utils.set_qflist(errors, { auto_open = config.auto_open_qflist, auto_close = config.auto_close_qflist })
+    utils.set_qflist(errors, {
+      auto_open = config.auto_open_qflist,
+      auto_close = config.auto_close_qflist,
+      auto_focus = not config.flags.watch,
```

Comment:
Nit: This seems probably too opinionated. Lets add a new configuration option, `auto_focus_qflist`, with the default value being `false` (though I'm open for arguments for it being `true`)